static/app/components/forms/formField/index.tsx

```diff
@@ -341,20 +336,26 @@ class FormField extends Component<FormFieldProps> {
               >
                 <Observer>
                   {() => {
-                    const error = this.getError();
-                    const value = model.getValue(name);
+                    const error = getError();
+
+                    // XXX(epurkhiser): It is possible to use the form field in
+                    // a controlled state. THIS IS NOT WELL SUPPORTED though
+                    // and isn't really recommended.
+                    //
+                    // For now we will leave the `value` type off FormFieldProps.
+                    const value = (props as any).value ?? model.getValue(name);
```

Comment:
This was pretty nasty before.

Previously every time we rendered we re-created the MockModel object and injected the current value from the props.